docs/contributors_guide/Contribution_Procedure.rst

```diff
@@ -1,50 +1,108 @@
+#. Search the `SIMP JIRA`_ for an open ticket that is relevant to the issue or
+   open a new one.
 
-   * Suggestion: name the branch after the issue (e.g., 'SIMP-999')
+#. Use the `GitHub GUI to fork and clone`_ the repository (we'll use
+   ``pupmod-simp-iptables`` for the rest of this walkthrough)
 
-#. Make your changes!
+#. Clone the repo you want to work on:
 
-   * SIMP contributions should observe the `Puppet Language Style Guide`_
-     conventions where feasible
-   * Contributions should ideally include relevant spec and/or acceptance tests
+   * ``git clone git@github.com:<YOUR_GITHUB_NAME>/pupmod-simp-iptables iptables``
+
+#. Change into the repository's directory:
+
+   * ``cd iptables``
+
+#. Create a `feature branch`_: ``git checkout -b SIMP-XXXX``
+
+#. Do your work! `(Including tests, of course)`
+
+#. Commit your work. We will `squash`_ your `pull request`_ into one commit
+   when we merge it, so you can use as many commits as you'd like.
+
+   .. IMPORTANT::
+      The **first** commit should use the `Commit Message Conventions`_
+
+#. Push your changes to Github on your feature branch:
+
+   * ``git push origin SIMP-XXXX``
+
+#. Using the GitHub GUI, create a `pull request`_ from your feature branch to the branch of the
+   original repo that you want to contribute to.
+
+#. `Travis-CI`_ will run the spec tests for the branch and a member of the SIMP
+   team will review your submission. You should receive emails from Github as
+   the code reviews progress.
+
+#. If changes are requested, read the section below on amending your pull
+   request.
+
+Commit Message Conventions
+--------------------------
+
+An example commit message that following the SIMP conventions:
 
-#. Save your changes in a **single commit**
+  .. code-block:: none
 
-   * Use the following commit message conventions:
+     (SIMP-999) Fix the broken thing [50 chars max]
 
-     .. code-block:: none
+     Discussion about the fix (if needed) [each line: 72 chars max]
 
-        (SIMP-999) Fix the broken thing [50 chars max]
+     SIMP-998 #comment Comment on a related issue [72 chars max]
+     SIMP-999 #close
 
-        Discussion about the fix (if needed) [each line: 72 chars max]
+The first commit message should be the following format:
 
-        SIMP-998 #comment Comment on a related issue [72 chars max]
-        SIMP-999 #close
+  * First line:
 
-   * The commit message should be the following format:
+    * Start with the Issue name in parentheses [e.g., ``(SIMP-999)``], followed
+      by a summary of the change
+    * No longer than **50** characters
+    * Followed by a line of white space
 
-     * First line:
+  * Subsequent lines:
 
-       * Start with the Issue name in parentheses [e.g., ``SIMP-999``],
-         followed by a summary of the change
-       * No longer than **50** characters
-       * Followed by a line of white space
+    * Each line should be no longer than **72** characters
```

Comment:
This should also be expanded to cover the substance of the message. "Use the body to explain what and why vs. how.," as Chris Beams puts it. Puppet has a good example [here](https://github.com/puppetlabs/puppet/blob/master/CONTRIBUTING.md).  @op-ct summed it up something like this:

> `<previous behavior> <why that is a problem> <details of what and how the change fixes the problem>`